### PR TITLE
Fixes #880. Allow visiting /activity/username route directly.

### LINKED ISSUE #880: Error when log in at /activity/magsout and no data provide
Not logged, go to https://webcompat.com/activity/magsout, log in.  There is a flash message about `API CALL NOT FOUND`   Label in dropdown is wrong (avatar user)

capture d ecran 2015-12-19 a 09 09 10

Plus, I cant's see my activity, but If I go to Homepage et back to the previous url, it's ok.

### PR DESCRIPTION
**Pull request by miketaylr**
r? @karlcow 

(sorry for taking so long here @magsout)

### CODE REVIEW

**Comment by miketaylr** [comment]
Let's not merge yet, I should write a test or two for this.

**Comment by karlcow** [comment]
r+ @miketaylr 

not to self. Need to refactor this errorhandler think. We are reusing the same code over and over again. 
Probably an opportunity for a function around this code:

```python
    if (request.path.startswith('/api/') and
       request.accept_mimetypes.accept_json and
       not request.accept_mimetypes.accept_html):
        message = {
            'status': 400,
            'message': 'API call. ' + message,
        }
        resp = jsonify(message)
        resp.status_code = 400
```

More something like (but to think not completely satisfied with the code below)

```python
ERROR_DICT = {
    400: 'Bad Request.',
    401: 'Unauthorized. Please log in.',
    403: 'Forbidden. Maybe that looking at private stuff?',
    404: 'Not Found. Lost in Punk Cat Space',
    }


@app.errorhandler(400)
@app.errorhandler(401)
@app.errorhandler(403)
@app.errorhandler(404)
def custom_error_handler(err):
    if api_call(request):
        return api_message(err.code)
    return render_template(
        'error.html',
        error_code=err.code,
        error_message=ERROR_DICT[err.code])


def api_call(request):
    '''Checks if it's an API call'''
    if (request.path.startswith('/api/') and
       request.accept_mimetypes.accept_json and
       not request.accept_mimetypes.accept_html):
        return True
    else:
        return False


def api_message(code):
    '''Prepares HTTP response for API calls.'''
    message = {
        'status': code,
        'message': ERROR_DICT[code],
    }
    resp = jsonify(message)
    resp.status_code = 404
    return resp
```

**Comment by miketaylr** [comment]
thanks @karlcow -- I agree, we could abstract the error handling into a nice little method. Will fix the failing nose test and write some new ones before merging.

**Comment by miketaylr** [comment]
OK, tests are passing -- let's merge. Thanks for review @karlcow.

File: webcompat/views.py [hunk 1]
```diff
@@ -278,6 +295,24 @@ def jumpship(e):
     return redirect(url_for('index'))
 
 
+@app.errorhandler(400)
+def unauthorized(err):
```
Comment (karlcow) [inline webcompat/views.py]: Ah my bad!!!!
Need better eyeballs.

@miketaylr The function is being redefined You need to change the name when you cut and paste :p

Comment (karlcow) [inline webcompat/views.py]: 3 times the same function :D for 400, 401 and 403

Comment (miketaylr) [inline webcompat/views.py]: ooops, will fix tomorrow!

Comment (miketaylr) [inline webcompat/views.py]: (also, thanks -- good catch, even if a day late :p)

**Pull request merged**